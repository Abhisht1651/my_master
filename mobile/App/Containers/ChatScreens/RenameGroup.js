/* eslint-disable react/jsx-props-no-spreading */
// @flow
import React, { useState, useEffect } from 'react'
import { View, Text, TouchableOpacity, Modal, Image, TextInput } from 'react-native'
import Images from '../../Themes/Images'
import Colors from '../../Themes/Colors'
import { MediumThinButton } from '../../Components/Buttons/MediumThinButton'

// Styles
import styles from './ScreenStyles/RenameGroupStyles'

type Props = {
  captureText: (arg: string) => void,
  isModalVisible: boolean,
  onBtnPress: () => void,
  onClose: () => void
}

const RenameGroup = (props: Props) => {
  const [value, setValue] = useState('')

  const { captureText, isModalVisible, onBtnPress, onClose } = props
  useEffect(() => {
    captureText(value)
  }, [value])

  const changeTextHandler = (text: string) => {
    setValue(text)
  }
  const textColorStyle = {
    color: '#000080'
  }
  return (
    <Modal visible={isModalVisible} transparent>
      <View style={styles.screen}>
        <View style={styles.innerScreen}>
          <View style={styles.view1}>
            <TouchableOpacity style={styles.closeIconContainer} activeOpacity={1} onPress={onClose}>
              <Image source={Images.closeIcon} style={styles.closeIcon} />
            </TouchableOpacity>
          </View>
          <View style={styles.view2}>
            <Text style={styles.headerText}>Rename Group</Text>
          </View>
          <View style={styles.innerContainer}>
            <TextInput
              placeholder="Write something..."
              multiline
              placeholderTextColor="#8F8FAF"
              selectionColor={Colors.cursor}
              maxLength={75}
              spellCheck={false}
              autoCorrect={false}
              value={value}
              style={[styles.innerContainerText, textColorStyle]}
              onChangeText={changeTextHandler}
            />
          </View>
          <View style={styles.view3}>
            <MediumThinButton title="Save" onPress={onBtnPress} />
          </View>
        </View>
      </View>
    </Modal>
  )
}
RenameGroup.defaultProps = {
  isModalVisible: true
}

export default RenameGroup
