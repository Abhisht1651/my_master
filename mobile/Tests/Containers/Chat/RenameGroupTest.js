import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity, Text, Image, TextInput } from 'react-native'
import RenameGroup from '../../../App/Containers/ChatScreens/RenameGroup'

describe('RenameGroup', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<RenameGroup />) // no compile-time error
  })
  it('RenameGroup component renders correctly', () => {
    const wrapper = shallow(<RenameGroup />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 1 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(1)
  })
  it('should render 1 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(1)
  })
  it('should render 1 Image', () => {
    expect(wrapper.find(Image)).toHaveLength(1)
  })
  it('should render 1 TextInput', () => {
    expect(wrapper.find(TextInput)).toHaveLength(1)
  })
})
