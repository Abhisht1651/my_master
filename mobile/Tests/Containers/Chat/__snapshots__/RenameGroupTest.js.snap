// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenameGroup RenameGroup component renders correctly 1`] = `
<Component
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2222234D",
        "height": 1334,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "height": 527.3571428571429,
          "width": 650,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#E6ECED",
            "flexDirection": "row",
            "height": 87.07142857142857,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <ForwardRef
          activeOpacity={1}
          style={
            Object {
              "marginLeft": "auto",
              "marginRight": 41.94,
            }
          }
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "aspectRatio": 1,
                "height": 17.28,
              }
            }
          />
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 30,
              "letterSpacing": -0.14,
            }
          }
        >
          Rename Group
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderBottomColor": "#E6ECED",
            "borderBottomWidth": 1,
            "borderRadius": 10,
            "marginTop": 77.21428571428571,
            "width": 546,
          }
        }
      >
        <Component
          allowFontScaling={true}
          autoCorrect={false}
          maxLength={75}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Write something..."
          placeholderTextColor="#8F8FAF"
          rejectResponderTermination={true}
          selectionColor="#7B46E4"
          spellCheck={false}
          style={
            Array [
              undefined,
              Object {
                "color": "#000080",
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 65.71428571428571,
            "marginTop": "auto",
          }
        }
      >
        <MediumThinButton
          disabled={false}
          onPress={[Function]}
          style={Object {}}
          title="Save"
        />
      </View>
    </View>
  </View>
</Component>
`;
